realm/realm-library/src/main/java/io/realm/internal/Context.java

```diff
@@ -17,127 +17,24 @@
-    // The Context object is used to store a list of native pointers 
-    // whose disposal need to be handed over from the garbage 
-    // collection thread to the users thread.
-
-    private List<Long> abandonedTables = new ArrayList<Long>();
-    private List<Long> abandonedTableViews = new ArrayList<Long>();
-    private List<Long> abandonedQueries = new ArrayList<Long>();
-
-    private ReferencesPool referencesPool = new ReferencesPool();
-    private ReferenceQueue<NativeObject> referenceQueue = new ReferenceQueue<NativeObject>();
-
-    private boolean isFinalized = false;
-
-    public synchronized void addReference(int type, NativeObject referent) {
-        referencesPool.add(new NativeObjectReference(type, referent, referenceQueue, referencesPool.getFreeIndex()));
-    }
-
-    public synchronized void executeDelayedDisposal() {
-        for (int i = 0; i < abandonedTables.size(); i++) {
-            long nativePointer = abandonedTables.get(i);
-            Table.nativeClose(nativePointer);
-        }
-        abandonedTables.clear();
-
-        for (int i = 0; i < abandonedTableViews.size(); i++) {
-            long nativePointer = abandonedTableViews.get(i);
-            TableView.nativeClose(nativePointer);
-        }
-        abandonedTableViews.clear();
-
-        for (int i = 0; i < abandonedQueries.size(); i++) {
-            long nativePointer = abandonedQueries.get(i);
-            TableQuery.nativeClose(nativePointer);
-        }
-        abandonedQueries.clear();
-
-        cleanNativeReferences();
-    }
-
-    private void cleanNativeReferences() {
-        NativeObjectReference reference = (NativeObjectReference) referenceQueue.poll();
-        while (reference != null) {
-            // Dealloc the native resources
-            reference.cleanup();
-            // Inline referencesPool.remove() to make it faster.
-            // referencesPool.pool.set(index, null); is not really needed. Make it faster by not
-            // setting the slot to null.
-            referencesPool.freeIndexList.add(reference.refIndex);
-            reference = (NativeObjectReference) referenceQueue.poll();
-        }
-    }
-
-    public void asyncDisposeTable(long nativePointer, boolean isRoot) {
-        if (isRoot || isFinalized) {
-            Table.nativeClose(nativePointer);
-        }
-        else {
-            abandonedTables.add(nativePointer);
-        }
-    }
-
-    public void asyncDisposeTableView(long nativePointer) {
-        if (isFinalized) {
-            TableView.nativeClose(nativePointer);
-        }
-        else {
-            abandonedTableViews.add(nativePointer);
-        }
-    }
-
-    public void asyncDisposeQuery(long nativePointer) {
-        if (isFinalized) {
-            TableQuery.nativeClose(nativePointer);
-        }
-        else {
-            abandonedQueries.add(nativePointer);
-        }
+    static {
+        finalizingThread.setName("RealmFinalizingDaemon");
```

Comment:
I have thought about that, but i don't really know what is benefit of daemon thread in Android. https://groups.google.com/forum/#!topic/android-developers/E67KirHZd3I It seems no difference on Android?